src/Artifacts.jl

```diff
@@ -186,6 +186,56 @@ function archive_artifact(hash::SHA1, tarball_path::String; honor_overrides::Boo
     end
 end
 
+"""
+    ArtifactDownloadInfo
+
+Auxilliary information about an artifact to be used with `bind_artifact!()` to give
+a download location for that artifact, as well as the hash and size of that artifact.
+"""
+struct ArtifactDownloadInfo
+    # URL the artifact is available at as a gzip-compressed tarball
+    url::String
+
+    # SHA256 hash of the tarball
+    hash::Vector{UInt8}
+
+    # Size in bytes of the tarball.  A size of `0` means unknown.
+    size::UInt64
```

Comment:
To prevent confusion with an empty file, `size` could be an `Int64` with `-1`  used to mean unknown. `size::Union{UInt64, Nothing}` would also work.

Reply:
Empty tarball files are not allowed.

Reply:
That makes sense.